### PR TITLE
Pin sphinx <7.0, docutils <0.18

### LINKED ISSUE #4177: [docs] core/topologyattrs page format is broken
## Expected behavior ##
All docs pages are well-formatted.

## Actual behavior ##
Doc page [11.3.3. Topology attribute objects — [MDAnalysis.core.topologyattrs](https://docs.mdanalysis.org/stable/documentation_pages/core/topologyattrs.html) is not formatted correctly (see screen shot, sorry, long)

- footer with authors on the side
- classes and functions are partially hideen by side menu and at end instead of integrated into page

image

## Code to reproduce the behavior ##

Load https://docs.mdanalysis.org/stable/documentation_pages/core/topologyattrs.html#MDAnalysis.core.topologyattrs.Masses.radius_of_gyration in desktop web browser

## Current version of MDAnalysis ##

- docs version 2.5.0

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #4177, #4194

Changes made in this Pull Request:
 - pinned sphinx <7.0
 - pinned docutils <0.18

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

----
:books: Documentation preview :books:: https://mdanalysis--4196.org.readthedocs.build/en/4196/

### CODE REVIEW

**Comment by RMeli** [comment]
I run into this too recently (https://github.com/RMeli/spyrmsd/pull/77); the RTD theme is not compatible with `sphinx>=7`: https://github.com/readthedocs/sphinx_rtd_theme/issues/1463.

**Comment by lilyminium** [comment]
Thanks @RMeli! Yeah I was surprised that RTD wasn't also failing with that error. Turns out it seems to defualt to sphinx 5.0.2 anyway: https://readthedocs.org/api/v2/build/21298747.txt and https://readthedocs.org/api/v2/build/21305166.txt. It's still suitably messed
 up: https://mdanalysis--4196.org.readthedocs.build/en/4196/documentation_pages/core/topologyattrs.html

I think I might have tracked it down to docutils updating to 0.18.1!

File: maintainer/conda/environment.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ dependencies:
   - chemfiles>=0.10
   - codecov
   - cython
+  - docutils <0.18
```
Comment (IAlibay) [inline maintainer/conda/environment.yml:9]: If the issue is docutils, is the sphinx pin also necessary? (or is sphinx not backwards compatible here?)

**Review by IAlibay — APPROVED** [review]
The one question but otherwise I'll approve.

File: maintainer/conda/environment.yml (hunk 1 above)
Comment (lilyminium) [reply maintainer/conda/environment.yml:9]: It's more of a cautionary pin -- we already know that sphinx_rtd_theme doesn't work with >=7. Up until now we've just happened to install sphinx<7 in both our CI (6.2.1 in PR 4198, 5.0.2 in RTD) but relying on that seemed fragile.

**Pull request merged**